### DISCUSSION TITLE
443 port redirection deep traffic analysis.

### DISCUSSION THREAD

**Discussion by uchiit** [nextcloud/all-in-one#3932] [Questions] [answered]
Well... 

After some investigation of decrypted traffic in Wireshark it looks like that when your browser gives empty GET request (without path to any file) NextCloud will respond to your browser with an HTTP2 status code 302 redirect message (even if you are already authenticated and have coockies).

But if you will try to send GET request with particular path (/apps/dashboard/) and if you already authenticated, in such case NextCloud will give you a status code 200 which will not make a redirect for you and home screen eventually will be provided to you. But at the same moment your browser will try to open additional sessions to port 443 to receive some other additional stuff which is required on this main page.

Here is some screenshots.

**First:** HTTP2 with 302 Status Code response from server on empty GET request from outside network browser on port 45135 (outside port of NextCloud on edge router)
![first request redirect message (w coockie)](https://github.com/nextcloud/all-in-one/assets/67806883/a3408dfb-3a26-4f4d-b3aa-65e59433a8ed)

**Second:** Browser view after 302 redirection Status code. Initial request was on port 45135 (my.domain.name:45135) without URI but as we can see after redirection there is no any port in URL (defaults to 443).
![after first redirect (w coockie)](https://github.com/nextcloud/all-in-one/assets/67806883/8b0119f9-20d4-4a8a-93e5-025276fce9e0)

**Third:** HTTP2 response from server on GET request with _/apps/dashboard/_ path
![second request 200 OK message (w coockie)](https://github.com/nextcloud/all-in-one/assets/67806883/32eee5b0-f2ad-4800-9dc8-9c4652d53297)

**Fourth:** Browser view after 200 OK Status code. Due to there was no redirection, port number is preserved and we can see the home page.
![second successfull request (w coockie)](https://github.com/nextcloud/all-in-one/assets/67806883/6066c274-9852-4f8b-87a6-c60f2c012820)

In both cases TCP port 45135 was used but behavior varies.

_Originally posted by @uchiit in https://github.com/nextcloud/all-in-one/discussions/3477#discussioncomment-7835406_

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one#are-other-ports-than-the-default-443-for-nextcloud-supported

**Comment by QingXuDw** [comment]
@szaimen Similar to the issue @uchiit encountered with in #3929 ,  433 and 80 were blocked in my network by the ISP for some stupid security reasons. Yes, it's stupid, but it is not something I could possibly change. So is there any way or workaround to change the default 443 port to a customed one?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#are-other-ports-than-the-default-443-for-nextcloud-supported
